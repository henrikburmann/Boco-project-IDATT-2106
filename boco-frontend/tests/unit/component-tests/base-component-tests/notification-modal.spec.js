import { mount } from "@vue/test-utils";
import NotificationModal from "@/components/BaseComponents/NotificationModal.vue";

describe("NotificationModal component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(NotificationModal, {
      //passing prop to component
      props: {
        visible: true,
        title: "String",
        message: "String",
      },
    });
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});
