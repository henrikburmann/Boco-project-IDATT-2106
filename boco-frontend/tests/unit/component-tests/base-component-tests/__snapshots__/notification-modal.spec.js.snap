// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationModal component renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <!-- Main modal -->
  <div
    class="fixed grid place-items-center bg-gray-600 bg-opacity-50 top-0 left-0 right-0 z-50 w-full overflow-x-hidden overflow-y-auto inset-0 h-full"
  >
    <div
      class="relative w-full h-full max-w-2xl p-4 md:h-auto"
    >
      <!-- Modal content -->
      <div
        class="relative bg-white rounded-lg shadow dark:bg-gray-700"
      >
        <!-- Modal header -->
        <div
          class="flex items-start justify-between p-4 border-b rounded-t dark:border-gray-600"
        >
          <h3
            class="text-xl font-semibold text-gray-900 dark:text-white"
          >
            String
          </h3>
          <!-- Close button -->
          <button
            class="text-gray-400 bg-transparent hover:bg-gray-200 hover:text-gray-900 rounded-lg text-sm p-1.5 ml-auto inline-flex items-center dark:hover:bg-gray-600 dark:hover:text-white"
          >
            <svg
              aria-hidden="true"
              class="w-5 h-5"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 18L18 6M6 6l12 12"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </button>
        </div>
        <!-- Modal body -->
        <div
          class="p-6 space-y-6"
        >
          <p
            class="text-base leading-relaxed text-gray-500 dark:text-gray-400"
          >
            String
          </p>
        </div>
      </div>
    </div>
  </div>
  
</div>
`;
